app/models/framework_response/collection.rb

```diff
@@ -0,0 +1,51 @@
+class FrameworkResponse
+  class Collection < FrameworkResponse
+    validates :value_text, absence: true
+    validate :validate_presence, on: :update, if: -> { framework_question.required && parent.nil? }
+    validate :validate_details_collection, on: :update, if: -> { response_details? }
+
+    def self.sti_name
+      'collection'
+    end
+
+    def value
+      value_json.presence || []
+    end
+
+    def value=(answer)
+      self.value_json =
+        if response_details?
+          details_collection(answer).to_a
+        else
+          self.value_json = answer.presence || []
+        end
+    end
+
+    def option_selected?(option)
+      value.map { |v| v['option'] }.include?(option)
+    end
+
+  private
+
+    def details_collection(collection)
+      DetailsCollection.new(
+        collection: collection,
+        question_options: framework_question.options,
+        details_options: framework_question.followup_comment_options,
+      )
+    end
+
+    def validate_presence
+      errors.add(:value_json, :blank) if value.empty?
+    end
+
+    def validate_details_collection
+      # TODO: Add proper validation messages
+      errors.add(:value, 'Details collection is invalid') if details_collection(value).invalid?
```

Comment:
Could these error messages be moved out to the locale file? Not essential but it's nice to keep the models free of these if possible.

Reply:
I was wondering this. I think we shouldn't since we don't support localisation at all currently and we'd be creating new patterns in the app that don't currently exist

Reply:
updated